src/components/wallet/views/claimableDisplay.tsx

```diff
@@ -0,0 +1,58 @@
+import { h } from '@stencil/core'
+import { has as loHas } from 'lodash-es'
+import WalletButton from './walletButton'
+import { Wallet } from '../wallet'
+
+interface ClaimableBalance {
+  id: string
+  asset: string
+  amount: string
+  sponsor?: string
+  claimants: any[]
+}
+
+export default function claimableDisplay(this: Wallet) {
+  if (this.account.availableBalances) {
+    const claimableRecords = (claimableBalances: any) => {
+      return claimableBalances.records.map(claimableBalanceRow)
+    }
+    const claimableBalanceRow = (balance: ClaimableBalance) => {
+      const loadingKey = (type: string) => {
+        return `${type}:${balance.asset.split(':')[0]}:${balance.id}`
+      }
+      const assetCode = balance.asset.split(':')[0]
+      const claimBalanceButton = WalletButton.call(
+        this,
+        'Claim',
+        loadingKey('claim'),
+        () => {
+          this.claimAsset(
```

Comment:
It seems to me that we could get away with only passing `balance`, `this.claimAsset(balance)`?